project/src/main/kotlin/org/ionproject/core/calendarTerm/CalendarTermRepoImpl.kt

```diff
@@ -12,13 +13,19 @@ class CalendarTermRepoImpl(
     private val classMapper: KlassReducedMapper
 ) : CalendarTermRepo {
 
-    override fun getTerms(page: Int, limit: Int): List<CalendarTerm> = tm.run { handle ->
-        handle.createQuery("SELECT * FROM dbo.CalendarTerm OFFSET :offset LIMIT :lim")
-            .bind("offset", page * limit)
-            .bind("lim", limit)
-            .map(calendarTermMapper)
-            .list()
-    } as List<CalendarTerm>
+    override fun getTerms(page: Int, limit: Int): List<CalendarTerm> {
+        val result = tm.run { handle ->
+            handle.createQuery("SELECT * FROM dbo.CalendarTerm OFFSET :offset LIMIT :lim")
+                    .bind("offset", page * limit)
+                    .bind("lim", limit)
+                    .map(calendarTermMapper)
+                    .list()
+        } as List<CalendarTerm>
+
+        if(result.isEmpty())
+            throw ResourceNotFoundException("No calendar-terms found. (Check if your page & limit are valid)")
```

Comment:
Why does an empty result throw an exception? If there are no calendar terms on the database the return should probably be an empty list right?